truffle/com.oracle.truffle.tools.test/src/com/oracle/truffle/tools/TruffleProfilerTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.truffle.tools;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.oracle.truffle.api.instrumentation.AbstractInstrumentationTest;
+import com.oracle.truffle.api.source.Source;
+import com.oracle.truffle.api.source.SourceSection;
+import com.oracle.truffle.tools.TruffleProfiler.Counter;
+import com.oracle.truffle.tools.TruffleProfiler.Counter.TimeKind;
+import com.oracle.truffle.tools.TruffleProfiler.TestHook;
+
+public class TruffleProfilerTest extends AbstractInstrumentationTest {
+
+    private TruffleProfiler profiler;
+
+    @Before
+    public void setupDebugger() throws IOException {
+        TruffleProfiler.setTestHook(new TestHook() {
+            public void onCreate(TruffleProfiler p) {
+                profiler = p;
```

Comment:
So, yes, this is a prototype, but I am wondering whether the use case I describe is generally relevant. At the moment, there does not seem to be way to get from a `PolyglotEngine.Instrument` to any actual tool instance.

For the dynamic execution metric instrument I want to implement, I want to get to the data and process it independent of the instrument.

You said somewhere else that instruments are completely isolated. Is this completely necessary? I think it might be desirable to get access to the tools for a specific polyglot engine. Is this a security concern or something similar?